### PR TITLE
feat: upgrade oxc

### PR DESCRIPTION
**Pull request by Boshen**
The next oxc version improves sourcemap performance, and fixes various codegen annoyances.

### CODE REVIEW

File: crates/rolldown/tests/esbuild/dce/dce_of_symbol_instances/artifacts.snap [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-source: crates/rolldown_testing/src/case/case.rs
-expression: content
+source: crates/rolldown_testing/src/integration_test.rs
+expression: snapshot
```
Comment (Boshen) [inline crates/rolldown/tests/esbuild/dce/dce_of_symbol_instances/artifacts.snap]: We can get rid of this line from the snapshot by a setting.

Comment (hyf0) [reply crates/rolldown/tests/esbuild/dce/dce_of_symbol_instances/artifacts.snap]: Yeah. It's quite annoying. I'll take a look.

**Comment by hyf0** [comment]
Run `just test-rust` local and the snapshot conflicts should be solved.

**Comment by hyf0** [comment]
> Run `just test-rust` local and the snapshot conflicts should be solved.

My bad. You probably need to rebase the main branch to apply those newest settings. You could run `just test-rust` in solving conflicts and not need to solve manully.

File: crates/rolldown_binding/src/types/binding_output_chunk.rs [hunk 2]
```diff
@@ -86,12 +86,7 @@ impl BindingOutputChunk {
 
   #[napi(getter)]
   pub fn map(&self) -> napi::Result<Option<String>> {
-    self
-      .inner
-      .map
-      .as_ref()
-      .map(|v| v.to_json_string().map_err(|e| napi::Error::from_reason(format!("{e:?}"))))
-      .transpose()
+    Ok(self.inner.map.as_ref().map(SourceMap::to_json_string))
```
Comment (Boshen) [inline crates/rolldown_binding/src/types/binding_output_chunk.rs:89]: This no longer returns `Result`, we can remove `Result` in a follow up PR.

**Comment by hyf0** [comment]
`just fmt` for fix format issues.

**Review by hyf0 — APPROVED** [review]
Nice

**Pull request merged**